data-serving/data-service/src/model/case.ts

```diff
@@ -1,43 +1,289 @@
 import mongoose from 'mongoose';
+import { ObjectId, Double } from 'mongodb';
 
-export const validOutcomes = ['pending', 'recovered', 'death'];
+export enum Sex {
+    Female,
+    Male,
+    Other,
+}
 
-const demographicsSchema = new mongoose.Schema({
-    age: {
+export enum Outcome {
+    Pending,
+    Recovered,
+    Death,
+}
+
+export enum Species {
+    HomoSapien,
+}
+
+export enum TravelPurpose {
+    Family,
+    Conference,
+    Work,
+}
+
+const dateValidator = {
+    type: Date,
+    min: '2019-11-01',
+    max: Date.now,
+};
+
+const rangeSchema = new mongoose.Schema({
+    minimum: {
+        type: Number,
+        min: -1,
+        max: 200,
+    },
+    maximum: {
         type: Number,
         min: 0,
         max: 200,
     },
+});
+
+const ageSchema = new mongoose.Schema({
```

Comment:
Sort of a higher-level comment: are there any trade-offs of defining every schema explicitly vs. using a nested schema?

https://mongoosejs.com/docs/subdocs.html#altsyntaxarrays
https://mongoosejs.com/docs/subdocs.html#altsyntaxsingle

Reply:
No personal preference -- just didn't know where to draw the line, so I did them all explicitly :) Do you have a preference?

Reply:
Nope. Just curious if you're aware of any material trade-offs.